torchvision/utils.py

```diff
@@ -121,10 +126,74 @@ def save_image(
             If a file object was used instead of a filename, this parameter should always be used.
         **kwargs: Other arguments are documented in ``make_grid``.
     """
-    from PIL import Image
     grid = make_grid(tensor, nrow=nrow, padding=padding, pad_value=pad_value,
                      normalize=normalize, range=range, scale_each=scale_each)
     # Add 0.5 after unnormalizing to [0, 255] to round to nearest integer
     ndarr = grid.mul(255).add_(0.5).clamp_(0, 255).permute(1, 2, 0).to('cpu', torch.uint8).numpy()
     im = Image.fromarray(ndarr)
     im.save(fp, format=format)
+
+
+def draw_bounding_boxes(
+    image: torch.Tensor,
+    boxes: torch.Tensor,
+    labels: torch.Tensor,
+    label_names: List[str] = None,
+    colors: Dict[int, str] = None,
+    draw_labels: bool = True,
+    width: int = 1
+) -> torch.Tensor:
+
+    """
+    Draws bounding boxes on given image.
+
+    Args:
+        image (Tensor): Tensor of shape (C x H x W) or (1 x C x H x W)
+        bboxes (Tensor): Tensor of size (N, 4) containing bounding boxes in (xmin, ymin, xmax, ymax) format.
+        labels (Tensor): Tensor of size (N) Labels for each bounding boxes.
+        label_names (List): List containing labels excluding background.
+        colors (dict): Dict with key as label id and value as color name.
+        draw_labels (bool): If True (default) draws label names on bounding boxes.
+        width (int): Width of bounding box.
+    """
+
+    # Code co-contributed by sumanthratna
+
+    if not (torch.is_tensor(image)):
+        raise TypeError(f'tensor expected, got {type(image)}')
+
+    if(image.dim() == 4):
```

Comment:
I would say to just support `image.dim() == 3`, as this makes the contract of the function easier to understand.

Also, in python we generally don't use parenthesis between conditionals and loops, can you remove those?

Reply:
The only rationale behind supporting `(1 x C x H x W) ` was since the user can pass `(1 x C x H x W)` to detection models to get outputs. The same input can be given to this utility as well.

We can change this though, do let me know.